README.md

```diff
@@ -1,47 +1,52 @@
 ```
 
-![Distributed Trace](doc/distributed-trace.png)
+with the information above we're able to know that a failure happened at `CallChildActivity()` method (line 22) which was called by the `Main()` method (line 11) both inside the `Program.cs` class. That's the reason why the runtime message tracking is essential for a software health and reliability, a rich information like that greatly increases the troubleshoot productivity. So the `stack trace` serves very well in terms of message trace when the subject is a single process application, like monolithic systems. On the other hand, when dealing with distributed systems like in a microservice architecture, the stack trace is not enough to expose the entire message tracking, that's the reason why distributed tracing tools and standards became necessary. The W3C defines a standard for this type of tracking, which is called `Trace Context`.
+
+## W3C Trace Context purpose
+
+Imagine a system designed as a microservice architecture where two APIs communicate in a synchronous way (http calls) and the second API communicates with a worker by a message broker:
 
-## APM tools/vendors
+### <a name="firstfigure"></a>Figure 1 - Distributed trace
 
-Application Performance Management tools examples:
+![distributed-trace](doc/distributed-trace.png)
+
+just like the `stack trace`, every `Activity` needs an Id to be identifiable and also needs to know the `Activity` Id of who called it. With the purpose of solve this kind of problem, some vendors came up delivering not only the distributed trace message information but also the application performance, the load time, the application's response time, and other stuff. That kind of vendor are called Application Performance Management tools (APM tools) or also trace systems, below are some examples:
 
 - Dynatrace
+- New Relic
 - Application Insights
 - Elastic APM
 - Zipkin
 
-## W3C Purpose
+> **_NOTE:_**  I chose to use the term `vendor` to describe all the trace systems because that's the way as the standard refers to them. But it seems that [it'll be changed soon](https://github.com/w3c/trace-context/issues/387).
 
-Defines standard HTTP headers and a value format to propagate context information that enables distributed tracing scenarios.
+now imagine a scenery where there are many vendors and also many languages with different diagnostics libraries, some of them identify a trace with an `operation-id`, other calls it as `request-id` and also there is another one which reconize it as a `trace-id`, besides that, the id's format changes depending on vendor or diagnostic library: one is in the `hierarchical` format, another one is an `UUID` and there is also a 24 character `string` identifier. That scenery would result in: systems with different tracing vendors will not be able to correlate the traces and also will not be able to propagate traces as there is no unique identification that is forwarded. This is where trace context standard comes in.
```

Comment:
**suggestion**: break this first sentence in two phrases:

```
...other calls it as `request-id` and also there is another one which reconize it as a `trace-id`. Besides that, the id's format...
```

Reply:
well noted, thanks!